### PR TITLE
bpo-29572: Update Windows build to OpenSSL 1.0.2k

### PR DESCRIPTION
**Pull request by zware**
`test_xmlrpc` fails, but it did before the update as well, so I'm calling it good.

### CODE REVIEW

File: Misc/NEWS [hunk 1]
```diff
@@ -806,6 +806,8 @@ Documentation
 Build
 -----
 
+- Issue #29572: Update Windows build and OS X installers to use OpenSSL 1.0.2k.
```
Comment (Mariatta) [inline Misc/NEWS]: I think the change here only reflect Windows build.
Is there going to be a separate PR for OS X?

Comment (zware) [reply Misc/NEWS]: Yes, that's up to @ned-deily.  I went ahead and copied the message from our 1.0.2j update assuming that it will be done anyway :)

Comment (ned-deily) [reply Misc/NEWS]: That's fine.  The additional PR is coming. Although aren't we using bpo- stylings now in Misc/NEWS?
